pymc3/distributions/distribution.py

```diff
@@ -356,43 +380,26 @@ def _draw_value(param, point=None, givens=None, size=None):
             return point[param.name]
         elif hasattr(param, 'random') and param.random is not None:
             return param.random(point=point, size=size)
+        elif (hasattr(param, 'distribution') and
+                hasattr(param.distribution, 'random') and
+                param.distribution.random is not None):
+            return param.distribution.random(point=point, size=size)
         else:
             if givens:
                 variables, values = list(zip(*givens))
             else:
                 variables = values = []
             func = _compile_theano_function(param, variables)
-            return func(*values)
+            if size and values and not all(var.dshape == val.shape for var, val in zip(variables, values)):
```

Comment:
I am getting `AttributeError: 'TransformedRV' object has no attribute 'dshape'` here.

I was testing with your model in the previous PR:
```python
with pm.Model() as model:
    phi = pm.Beta('phi', alpha=1., beta=1.)

    kappa_log = pm.Exponential('logkappa', lam=5.)
    kappa = pm.Deterministic('kappa', tt.exp(kappa_log))

    thetas = pm.Beta('thetas', alpha=phi*kappa, beta=(1.0-phi)*kappa, shape=N)

    y = pm.Binomial('y', n=at_bats, p=thetas, shape=N, observed=hits)
```

Reply:
I guess we can just add dshape to `TransformedRV`? Cant see why not.
```python
self.dshape = tuple(distribution.shape)
self.dsize = int(np.prod(distribution.shape))
```

Reply:
Thanks! That fixes it (added the example to the tests, too).